ref/README.md

```diff
@@ -1,56 +1,153 @@
+module (e.g. `module load git`).  If you are using a system that uses modules, use
+`module avail` to look for alternative versions of git that may have git-lfs support.
+
+Make sure the files in `data/inputs` are NetCDF data files (not text) before proceeding to
+the build step.
+
+**NOTE**: If you cloned the repository with a version of git without git-lfs installed, you
+must run the following command (with a version of git that does support git-lfs) from the base
+of the repository to fetch the input data before proceeding to the build steps. Or you can
+reclone the repository with git-lfs installed, instead.
+
+```bash
+$ git lfs pull
+```
 
 ## Building the kernel
 
 This kernel uses an out-of-source cmake build, meaning that the build must be done in 
 directory that is not in the source tree.
 
-The build procedure is (from the directory containing this file):
+### Basic build procedure (from the directory containing this file)
+
+The basic build steps are as follows (**NOTE**: Make sure not to omit the two dots at the end
+of the `cmake` step.):
+
+```bash
+$ rm -rf build ; mkdir build ; cd build
+$ export CC=<name of C compiler>
+$ export FC=<name of fortran compiler> 
+$ cmake -DCMAKE_BUILD_TYPE=<debug | release> ..
+$ make VERBOSE=1
+```
+
+On many systems, the above will suffice. However, some systems will require you to help cmake
+find dependencies, particularly if software depencencies are not installed in standard locations.
+See below for more information.
+
+### Machines that use modules to manage software
 
+Most HPC systems use modules to manage software.  Make sure you have loaded the versions of
+the compiler and software you want to use before running the build steps above.  This will allow build
+dependencies to be found properly.  For example:
+
+```bash
+$ module load intel netcdf cmake
 ```
-rm -rf build ; mkdir build ; cd build
-module load <compiler> netcdf cmake
-export CC=<gcc | icc>
-export FC=<gfortran | ifort>
-cmake -DCMAKE_BUILD_TYPE=<debug | release> ..
-make VERBOSE=1
+
+### Machines that do not use modules to manage software
+
+If compilers and/or NetCDF is not installed in a standard location where cmake can find it, you
+may need to add their installation paths to the `CMAKE_PREFIX_PATH` before running the steps
+above. For example:
+
+```bash
+$ export CMAKE_PREFIX_PATH=$CMAKE_PREFIX_PATH:/path/to/netcdf:/path/to/netcdf-fortran
+```
+
+### Building on a Mac
+
+By default, gcc points to the clang compiler on Mac.  To use the GNU compiler on Mac, depending
+on how the GNU compiler was installed, you may need to specify the C compiler name as gcc-$version.
+For example:
+
+```bash
+$ export CC=gcc-10
 ```
 
 ## Testing the kernel
 
-To run the test suite (from the build directory):
+First, set the OpenMP variable to 1. `yppm` is single threaded and this variable is not used. For example: 
 
-`ctest`
+```bash
+$ export OMP_NUM_THREADS=1
+```
+
```

Comment:
To avoid confusion perhaps this could be rephrased to explain that "Although `yppm` is  currently single-threaded, it may be threaded in the future, and setting `OMP_NUM_THREADS=1` avoids extraneous output of thread affinity information.

Reply:
That is quite a bit more articulate and explanatory!  I will take out the "For example" in this section as well.